Integrations/src/main/docker/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM deephaven/java-and-python:local-build as install-wheels
+
+RUN set -eux; \
+    apt-get -qq update; \
+    apt-get -qq -y --no-install-recommends install liblzo2-2 curl; \
+    rm -rf /var/lib/apt/lists/*
+
+COPY deephaven-jpy-wheel/ /deephaven-jpy-wheel
```

Comment:
why not do just one big copy here, then one big run, install all three wheels in one step? I expect that jpy itself changes less often than the other two wheels, but do you think its enough to warrant tripling the layers? 

The concern I have is that since old deephaven-wheel is based in part on our javadoc, it will see a change any time the JSON is correctly updated, so we will have to copy it, install it, copy the dh2-wheel, install it (even if it didnt change) - might as well just do them in one step?

("yes, i meant it this way" is a totally valid answer, just trying to make sure i get it all)

Reply:
So, we are "tripling" the layers, but it only effects the build stage, and not the final produced image. In essence, I think we are able to get the "best of both worlds" - we don't have to re-install deephaven-jpy.whl when deephaven.whl or deephaven2.whl changes.

Reply:
it also affects the local build cache, 2x the image churn - and deephaven.whl is the "big" one, relative to the other two (1MB, 50kB, 217MB) due to downloading other packages from the internet. On the one hand it should be first so we don't churn it so much, on the other hand, afaict it is the only one that changes when javadoc changes, so it should be last to avoid rebuilding the others.

And the final layer gets a complete fresh copy made when anything else changes.

Reply:
I'm happy to change the order of the COPY/RUN steps, or move into a single COPY/RUN.

But this also brings up another "reproducibility" concern very similar to your concern about apt-get installing specific versions. pip will likely choose the most recent version for all of the dependencies the wheel specifies. Our requirements *aren't* specific:

```
$ unzip -q -c docker/runtime-base/build/context/deephaven-wheel/deephaven-0.7.0-py2.py3-none-any.whl | grep Requires

Requires-Dist: deephaven-jpy (==0.7.0)
Requires-Dist: numpy
Requires-Dist: dill (>=0.2.8)
Requires-Dist: wrapt
Requires-Dist: pandas
Requires-Dist: enum34 ; python_version < "3.4"
Requires-Dist: numba ; python_version > "3.0"
```

If we want to maximize cache-ability, we would likely want to have a freeze list that we install before any of our wheels; and then can install just our wheels w/o any additional download.

Reply:
I've added a requirements file into the runtime-base PR - which in addition to being better for reproducibility, aids a bunch in cachability.